### PR TITLE
CHE-3585 Word Wrap Toggle for Workspace Editor

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?
Adds an option to `Profile > Preferences > Editor` to set soft wrap enabled by default.
Adds `Edit > Soft Wrap` menu item to enable or disable soft wrap for the current editor.

soft-wrap

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/3585

#### Changelog
An option to configure `Soft Wrap` is placed to `Profile > Preferences > Editor`.
Add option to `Edit > Soft Wrap` to toggle soft wrap for the current editor.

#### Release Notes

Editors are now getting an option to enable soft wrap capability so that it wraps the text that is getting larger that the editor pane. 
This option can be activate for a single file, by going into the menu `Edit > Word Wrap`. You can also default that behaviour for all your editors, by configuring your preferences: 
`Profile > Preferences > Editor` and enabling soft wrap option.

#### Docs PR

https://github.com/eclipse/che-docs/pull/184

### CODE REVIEW

**Comment by slemeur** [comment]
Release Note updated.

Please @vitaliy-guliy record a new animated gif with the new labels and by zooming in a little bit more. Also please add the proper changelog.

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/preferences/editorproperties/sections/EditPropertiesSection.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+@Singleton
+public class EditPropertiesSection implements EditorPropertiesSection {
+    private final List<String>                properties;
+    private final EditorLocalizationConstants locale;
+
+    @Inject
+    public EditPropertiesSection(EditorLocalizationConstants locale) {
+        this.locale = locale;
+        properties = Arrays.asList(
+                                   SOFT_WRAP.toString()
```
Comment (vparfonov) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/preferences/editorproperties/sections/EditPropertiesSection.java]: formater

**Review by JamesDrummond — APPROVED** [review]

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/preferences/editorproperties/sections/EditPropertiesSection.java (hunk 1 above)
Comment (vitaliy-guliy) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/preferences/editorproperties/sections/EditPropertiesSection.java]: Done

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/texteditor/CanWrapLines.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.api.editor.texteditor;
+
+/**
+ * Determined whether the editor implementing this interface can wrap lines.
```
Comment (ashumilova) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/texteditor/CanWrapLines.java:14]: author is missing

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/SoftWrapAction.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.google.inject.Inject;
+import org.eclipse.che.ide.CoreLocalizationConstant;
+import org.eclipse.che.ide.api.action.ActionEvent;
+import org.eclipse.che.ide.api.action.ToggleAction;
+import org.eclipse.che.ide.api.editor.EditorAgent;
+import org.eclipse.che.ide.api.editor.EditorPartPresenter;
+import org.eclipse.che.ide.api.editor.texteditor.CanWrapLines;
+
+/**
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/SoftWrapAction.java:21]: author is missing

**Review by ashumilova — APPROVED** [review]

**Review by vparfonov — APPROVED** [review]

**Review by slemeur — APPROVED** [review]
:+1:

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/texteditor/CanWrapLines.java (hunk 2 above)
Comment (vitaliy-guliy) [reply ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/texteditor/CanWrapLines.java:14]: Added

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/SoftWrapAction.java (hunk 3 above)
Comment (vitaliy-guliy) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/SoftWrapAction.java:21]: Added

**Comment by JamesDrummond** [comment]
Was really looking forward to this one in 5.6.0 but I guess I can wait for 5.7.0 :(

**Pull request merged**